### ISSUE TITLE
ffmpeginput.cpp fails, using old PIX_FMT_RGB24

### ISSUE THREAD

**Issue by cdc100** [OpenImageIO/oiio#1341]
The build, current master at least, fails when using FFmpeg
```
/usr/local/include/libavcodec/avcodec.h:4883:5: note: declared here
 int avpicture_fill(AVPicture *picture, const uint8_t *ptr,
     ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:374:21: error: ‘PIX_FMT_RGB24’ was not declared in this scope
                     PIX_FMT_RGB24,
                     ^
```
PIX_FMT_RGB24 and other entries were moved to [old_pix_fmts.h](https://github.com/FFmpeg/FFmpeg/blob/78071a1420b425dfb787ac739048f523007b8139/libavutil/old_pix_fmts.h) for FFMPEG 1.1 quite some time ago with https://github.com/FFmpeg/FFmpeg/commit/78071a1420b425dfb787ac739048f523007b8139

Is this feature deprecated or is it really required for FFmpeg in general to be such an old version?
A simple fix would replace the entries
```
--- ffmpeginput.cpp	2016-02-04 13:48:29.670077361 +0800
+++ ffmpeginput.cpp_orig	2016-02-04 13:48:11.513955458 +0800
@@ -245,7 +245,7 @@
     m_frame = av_frame_alloc();
     m_rgb_frame = av_frame_alloc();
     m_rgb_buffer.resize(
-        avpicture_get_size (AV_PIX_FMT_RGB24,
+        avpicture_get_size (PIX_FMT_RGB24,
         m_codec_context->width,
         m_codec_context->height),
         0
@@ -273,7 +273,7 @@
         pixFormat,
         m_codec_context->width,
         m_codec_context->height,
-        AV_PIX_FMT_RGB24,
+        PIX_FMT_RGB24,
         SWS_AREA,
         NULL,
         NULL,
@@ -371,7 +371,7 @@
                 (
                     reinterpret_cast<AVPicture*>(m_rgb_frame),
                     &m_rgb_buffer[0],
-                    AV_PIX_FMT_RGB24,
+                    PIX_FMT_RGB24,
                     m_codec_context->width,
                     m_codec_context->height
                 );
```

This leaves however several warnings for deprecated functions still.
```
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:248:9: warning: ‘int avpicture_get_size(AVPixelFormat, int, int)’ is deprecated [-Wdeprecated-declarations]
         avpicture_get_size (AV_PIX_FMT_RGB24,
         ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4898:5: note: declared here
 int avpicture_get_size(enum AVPixelFormat pix_fmt, int width, int height);
     ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:248:9: warning: ‘int avpicture_get_size(AVPixelFormat, int, int)’ is deprecated [-Wdeprecated-declarations]
         avpicture_get_size (AV_PIX_FMT_RGB24,
         ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4898:5: note: declared here
 int avpicture_get_size(enum AVPixelFormat pix_fmt, int width, int height);
     ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:250:32: warning: ‘int avpicture_get_size(AVPixelFormat, int, int)’ is deprecated [-Wdeprecated-declarations]
         m_codec_context->height),
                                ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4898:5: note: declared here
 int avpicture_get_size(enum AVPixelFormat pix_fmt, int width, int height);
     ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp: In member function ‘void OpenImageIO::v1_7::FFmpegInput::read_frame(int)’:
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:370:17: warning: ‘int avpicture_fill(AVPicture*, const uint8_t*, AVPixelFormat, int, int)’ is deprecated [-Wdeprecated-declarations]
                 avpicture_fill
                 ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4883:5: note: declared here
 int avpicture_fill(AVPicture *picture, const uint8_t *ptr,
     ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:370:17: warning: ‘int avpicture_fill(AVPicture*, const uint8_t*, AVPixelFormat, int, int)’ is deprecated [-Wdeprecated-declarations]
                 avpicture_fill
                 ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4883:5: note: declared here
 int avpicture_fill(AVPicture *picture, const uint8_t *ptr,
     ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:377:17: warning: ‘int avpicture_fill(AVPicture*, const uint8_t*, AVPixelFormat, int, int)’ is deprecated [-Wdeprecated-declarations]
                 );
                 ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4883:5: note: declared here
 int avpicture_fill(AVPicture *picture, const uint8_t *ptr,
     ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:389:17: warning: ‘void av_free_packet(AVPacket*)’ is deprecated [-Wdeprecated-declarations]
                 av_free_packet (&pkt);
                 ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4040:6: note: declared here
 void av_free_packet(AVPacket *pkt);
      ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:389:17: warning: ‘void av_free_packet(AVPacket*)’ is deprecated [-Wdeprecated-declarations]
                 av_free_packet (&pkt);
                 ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4040:6: note: declared here
 void av_free_packet(AVPacket *pkt);
      ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:389:37: warning: ‘void av_free_packet(AVPacket*)’ is deprecated [-Wdeprecated-declarations]
                 av_free_packet (&pkt);
                                     ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4040:6: note: declared here
 void av_free_packet(AVPacket *pkt);
      ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:393:9: warning: ‘void av_free_packet(AVPacket*)’ is deprecated [-Wdeprecated-declarations]
         av_free_packet (&pkt);
         ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4040:6: note: declared here
 void av_free_packet(AVPacket *pkt);
      ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:393:9: warning: ‘void av_free_packet(AVPacket*)’ is deprecated [-Wdeprecated-declarations]
         av_free_packet (&pkt);
         ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4040:6: note: declared here
 void av_free_packet(AVPacket *pkt);
      ^
/home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:393:29: warning: ‘void av_free_packet(AVPacket*)’ is deprecated [-Wdeprecated-declarations]
         av_free_packet (&pkt);
                             ^
In file included from /usr/local/include/libavformat/avformat.h:318:0,
                 from /home/cdc100/deploy/oiio/src/ffmpeg.imageio/ffmpeginput.cpp:33:
/usr/local/include/libavcodec/avcodec.h:4040:6: note: declared here
 void av_free_packet(AVPacket *pkt);
      ^
```

**Comment by sopvop** [comment]
#1359 should fix it

**Comment by cdc100** [comment]
This can be closed since it was addressed in #1359
